### PR TITLE
Feature/137005175/notification

### PR DESCRIPTION
**Pull request by olawalejarvis**
### What does this PR do?

Send SMS alert to buyers  for products ordered, and wallet activities.
Display notifications on the user dashboard.

### How should this be manually tested?
Register a user and place an order
Deposit money into wallet account
Pay for order from wallet account

What are the relevant pivotal tracker stories?

#137005175

### Screenshot(s)
oie_sg7xgzfavbrl

### CODE REVIEW

**Review by andela-aolaniran — APPROVED** [review]
Looks good

File: imports/plugins/core/checkout/client/templates/checkout/payment/methods/cards.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ Template.walletPayment.events({
         Meteor.call("cart/submitPayment", paymentMethod);
         Alerts.toast("Payment Successful", "success");
       } else {
-        Alerts.toast("An error occured, please try again", "error");
+        Alerts.toast("An error occured, please try again, I am the wallet", "error");
```
Comment (vonvick) [inline imports/plugins/core/checkout/client/templates/checkout/payment/methods/cards.js]: is there a reason for "I am the wallet"?

File: server/methods/core/orders.js [hunk 2]
```diff
@@ -353,10 +390,95 @@ Meteor.methods({
...
+      notification.name = "Order Completed";
+      notification.type = "completed";
+      notification.message = "Your order has been Delivered!";
+    }
+
+    // canceled order
+    if (order.workflow.status === "canceled") {
+      notification.name = "Order Canceled";
+      notification.type = "canceled";
+      notification.message = "Your order has been cancel!";
```
Comment (vonvick) [inline server/methods/core/orders.js]: would be nice if the cancel is spelt correctly

File: server/methods/core/wallet.js [hunk 3]
```diff
@@ -28,18 +38,44 @@ Meteor.methods({
         if (!recipient) {
           return 2;
         }
+        notification = {
+          userId: userId,
+          name: "Debit Transaction",
+          type: transactionType,
+          message: `Debit Alert! ${amount} has been transfer from your account to ${recipient._id}`,
```
Comment (vonvick) [inline server/methods/core/wallet.js]: the past tense of transfer would be appropriate here.

File: server/methods/core/wallet.js [hunk 4]
```diff
@@ -63,7 +99,17 @@ Meteor.methods({
     const orderId = orderInfo._id;
     userId = orderInfo.userId;
     const transaction = {amount, orderId, transactionType: "Refund", date: orderInfo.updatedAt};
+    const notification = {
+      userId: userId,
+      name: "Money Refund",
+      type: "Refund",
+      message: `Refund! ${amount} has be refunded into your account based on canceled order ${orderId}`,
```
Comment (vonvick) [inline server/methods/core/wallet.js]: "been" would be nice as against "be"

**Review by vonvick — COMMENTED** [review]
LGTM except for the few nitpicks

**Pull request merged**